tests/functional/oauth_choose_redirect.js

```diff
@@ -0,0 +1,82 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'intern',
+  'intern!object',
+  'intern/chai!assert',
+  'intern/node_modules/dojo/node!xmlhttprequest',
+  'app/bower_components/fxa-js-client/fxa-client',
+  'tests/lib/helpers',
+  'tests/functional/lib/helpers'
+], function (intern, registerSuite, assert, nodeXMLHttpRequest, FxaClient, TestHelpers, FunctionalHelpers) {
+  var config = intern.config;
+  var AUTH_SERVER_ROOT = config.fxaAuthRoot;
+  var CONTENT_SERVER_ROOT = config.fxaContentRoot;
+  var oAuthUrl = CONTENT_SERVER_ROOT + 'oauth?client_id=dcdb5ae7add825d2&scope=profile';
+  var PASSWORD = 'password';
+  var email;
+  var client;
+
+  registerSuite({
+    name: 'oauth choose redirect',
+
+    beforeEach: function () {
+      client = new FxaClient(AUTH_SERVER_ROOT, {
+        xhr: nodeXMLHttpRequest.XMLHttpRequest
+      });
+
+      return FunctionalHelpers.clearBrowserState(this);
+    },
+
+    'oauth endpoint redirects to signup with invalid account email': function () {
+      var self = this;
+
+      email = TestHelpers.createEmail();
+
+      var invalidAccountUrl = oAuthUrl + '&email=' + email;
+
+      return self.remote
```

Comment:
Same comments as below apply to this section, you should be able to:

```js
return FunctionalHelpers.openPage(this, invalidAccountUrl, '#fxa-signup-header')
  .then(FunctionalHelpers.testEelementValueEquals('input[type=email]', email));
```

To reduce the `FunctionalHelpers` references within the tests, I sometimes [create local references at the top of the file](https://github.com/mozilla/fxa-content-server/blob/0bba3fbb5770872a81efca2e85958416a056e00a/tests/functional/oauth_query_param_validation.js#L23:L27).